keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,849 @@
+    |------------------------------------------|       |   "name": "s3.amazonaws.com"          |
+                                                       | }                                     |
+                                                       |---------------------------------------|
+```
 
-    string principal = 3;
-    
-    // This field is REQUIRED
-    string access_policy = 4;
-}
+#### ProvisionerCreateBucket
 
-message ProvisionerGrantBucketAccessResponse {
-    // This is the account that is being provided access. This will
-    // be required later to revoke access. 
-    string principal = 1;
-    
-    string credentials_file_contents = 2;
-    
-    string credentials_file_path = 3;
-} 
+This gRPC call creates a bucket in the OSP, and returns information about the new bucket. This api must be idempotent. The input to this call is the name of the bucket and an opaque parameters field.
+
+The returned `bucketID` should be a unique identifier for the bucket in the OSP. This value could be the name of the bucket too. This value will be used by COSI to make all subsequent calls related to this bucket.
+
+```
+    ProvisionerCreateBucket
+    |------------------------------------------|       |-----------------------------------------------|
+    | grpc ProvisionerCreateBucketRequest{     | ===>  | ProvisionerCreateBucketResponse{              |
+    |     "name": "br-$uuid",                  |       |   "bucketID": "br-$uuid",                     |
+    |     "parameters": {                      |       |   "bucketInfo": {                             |
+    |        "key": "value"                    |       |      "s3": {                                  |
+    |     }                                    |       |        "bucketName": "br-$uuid",              |
+    | }                                        |       |        "region": "us-west1",                  |
+    | -----------------------------------------|       |        "endpoint": "s3.amazonaws.com"         |
+                                                       |      }                                        |
+                                                       |   }                                           |
+                                                       | }                                             |
+                                                       |-----------------------------------------------|
 ```
 
-## ProvisionerRevokeBucketAccess
+#### ProvisionerGrantBucketAccess
 
-This call revokes all access to a particular bucket from a principal.  
+This gRPC call creates a set of access credentials for a bucket. This api must be idempotent. The input to this call is the id of the bucket, a set of opaque parameters and name of the account. This `accountName` field is used to ensure that multiple requests for the same BucketClaim do not result in multiple credentials.
```

Comment:
How does the sidecar determine the `accountName`? Is this able to to support federated identity providers?

Reply:
the driver is expected to return the `accountName`. Since it is just a string that uniquely identifies the account owner, it is flexible enough to refer to any user in any IDP.

Reply:
We are passing in `accountName` in the arguments for this call. Where does that come from?

Reply:
still open

Reply:
`accountName` is the concatenation of the characters `ba` (short for BucketAccess) and its UID. It is used as the idempotency key for requests to the drivers regarding a particular BA. 

my earlier comment was about `accountID`, which the driver provides after successfully granting access. It was not about `accountName` - I made a mistake there https://github.com/kubernetes/enhancements/pull/2813/files#r809846548

Reply:
Thanks, can you explicitly state that in here? "$uuid" makes it seem like it's the same variable for br and ba